lib/client.js

```diff
@@ -256,15 +270,15 @@ function Client(opt, callback) {
   this.sock.pipe(this.input).pipe(this.consumer);
   this.output.pipe(this.sock);
   this.send(0, {cmd: 'connect', access: opt.access, data: opt.data}, function (err, data) {
-    var i, d;
     if (err) throw err;
     if (data) {
       self.id = data.id;
       self.uuid = data.uuid;
       self.emit('connect', data);
-      if (data.devices) {   // Cache remote ids
-        for (i = 0; i < data.devices.length; i++) {
-          d = data.devices[i];
+
+      // Cache remote ids
+      if (data.devices) {
+        for (const d in data.devices) {
```

Comment:
Is it correct ? Maybe you confuse with array.forEach.
Arrays should be almost always iterated with a numerical index. See this [jsperf](https://jsperf.com/fast-array-foreach)

To restrict the scope of local iterator, we should use ECMA 5.1 `let` statement. It should be something like:

```javascript
for (let i = 0; i < data.devices.length; i++) {
  let d = data.devices[i];
  ...
}
````